nifi/Dockerfile

```diff
@@ -4,74 +4,51 @@
-      vendor="Stackable GmbH" \
-      version="${PRODUCT}" \
-      release="${RELEASE}" \
-      summary="The Stackable image for Apache NiFi." \
-      description="This image is deployed by the Stackable Operator for Apache NiFi."
 
 RUN microdnf update && \
     microdnf install \
-    gcc \
-    gzip \
     libxml2-devel \
-    libxslt-devel \
-    tar \
-    unzip \
-    zip && \
-    microdnf install \
-    python-devel \
-    python-pip \
-    python-setuptools && \
+    libxslt-devel && \
     microdnf clean all \
     && rm -rf /var/cache/yum
 
-RUN pip install --no-cache-dir nipyapi==0.19.1
+COPY nifi/licenses /licenses
 
 # The bcrypt tool is needed by NiFi to locally encrypt the admin password that is mounted as a secret in cleartext
 # The source code is available at https://github.com/stackabletech/stackable-utils/tree/main/stackable-bcrypt
 # The binaries are published to Nexus with: mvn clean deploy
 # This jar used to be COPYed into the image from a local binary but it's not possible to do so anymore
 # because /bin is owned by root and curl cannot write there unless it runs as root.
 # This should be fixed by moving the jar to location under /stackable and updating the Nifi operator accordingly.
-RUN curl --fail -L 'https://repo.stackable.tech/repository/m2/tech/stackable/nifi/stackable-bcrypt/1.0-SNAPSHOT/stackable-bcrypt-1.0-20240508.153334-1-jar-with-dependencies.jar' -o /bin/stackable-bcrypt.jar \
-    && chown stackable:stackable /bin/stackable-bcrypt.jar
+RUN curl --fail -L 'https://repo.stackable.tech/repository/m2/tech/stackable/nifi/stackable-bcrypt/1.0-SNAPSHOT/stackable-bcrypt-1.0-20240508.153334-1-jar-with-dependencies.jar' \
+    # This used to be located in /bin/stackable-bcrypt.jar we create a softlink in the main container for backwards compatibility
+    -o /stackable/stackable-bcrypt.jar && \
+    chown stackable:stackable /stackable/stackable-bcrypt.jar
 
 USER stackable
 WORKDIR /stackable
 
-COPY --chown=stackable:stackable nifi/stackable /stackable
-COPY --chown=stackable:stackable nifi/licenses /licenses
-COPY --chown=stackable:stackable nifi/python /stackable/python
-
-
-# zip is different than tar and cannot be just piped, therefore the intermediate save and remove step to unzip
-RUN curl --fail -L https://repo.stackable.tech/repository/packages/nifi/nifi-${PRODUCT}-bin.zip -o /stackable/nifi-${PRODUCT}-bin.zip && \
-    unzip /stackable/nifi-${PRODUCT}-bin.zip && \
-    rm /stackable/nifi-${PRODUCT}-bin.zip && \
-    ln -s /stackable/nifi-${PRODUCT} /stackable/nifi
+RUN curl --fail -L "https://repo.stackable.tech/repository/packages/nifi/nifi-${PRODUCT}-source-release.zip" -o "/stackable/nifi-${PRODUCT}-source-release.zip" && \
+    unzip "nifi-${PRODUCT}-source-release.zip" && \
+    rm -rf "nifi-${PRODUCT}-source-release.zip" && \
+    mv nifi-${PRODUCT} nifi-${PRODUCT}-src && \
+    cd /stackable/nifi-${PRODUCT}-src/ && \
+    mvn clean install -DskipTests && \
```

Comment:
Not sure if this builds a real distribution like the one you download from their site. What you want here is some sort of dist profile or helper script that bundles not just java jars but also any scripts and what not.

Also it looks like the [skip-unit-tests](https://github.com/apache/nifi/blob/ea48d22fba666cc5ec70db70ccd4a13da07acff8/pom.xml#L1023) profile might be useful.

Reply:
It works, clean install is fine but i can check to opimize a little. https://nifi.apache.org/docs/nifi-docs/html/developer-guide.html#standard-build-instructions